xiaogpt/xiaogpt.py

```diff
@@ -180,6 +191,41 @@ async def do_tts(self, value, wait_for_finish=False):
                 await asyncio.sleep(2)
             print("回答完毕")
 
+    def start_http_server(self):
+        # set the port range
+        port_range = range(8050, 8090)
+        # get a random port from the range
+        self.port = random.choice(port_range)
+        # create the server
+        handler = http.server.SimpleHTTPRequestHandler
+        httpd = ThreadedHTTPServer(("", self.port), handler)
+        # start the server in a new thread
+        server_thread = threading.Thread(target=httpd.serve_forever)
+        server_thread.daemon = True
+        server_thread.start()
+
+        # local ip
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        s.connect(("8.8.8.8", 80))
+        self.local_ip = s.getsockname()[0]
+        s.close()
+        print(f"Serving on {self.local_ip}:{self.port}")
+
+    def text2mp3(self, value):
+        print(f"text2mp3 ")
+        cmd = f'edge-tts --text "{value}" --write-media output.mp3 --voice {self.config.edge_tts_voice}'
+        subprocess.run(cmd, shell=True)
+
+    async def play_url(self, url):
+        url = url or f"http://{self.local_ip}:{self.port}/output.mp3"
+        print(f"play: {url}")
+        await self.mina_service.play_by_url(self.device_id, url)
+
+    async def edge_tts(self, value):
+        print(f"edge_tts")
+        self.text2mp3(value)
+        await self.play_url("")
```

Comment:
I'd prefer this to be more explicit, the empty string doesn't make sense here. Why not return a file name by `text2mp3` and pass it to `play_url`?

Reply:
make sense

Reply:
updated.